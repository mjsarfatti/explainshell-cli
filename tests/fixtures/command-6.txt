tar zcf - some-dir | ssh some-server "cd /; tar xvzf -"