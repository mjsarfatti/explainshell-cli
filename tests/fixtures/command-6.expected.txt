tar

    The GNU version of the tar archiving utility

    [ tar -> https://explainshell.com/explain/1/tar ]

z

    -z, --gzip, --gunzip --ungzip

c

    -c, --create
          create a new archive

f -

    -f, --file ARCHIVE
          use archive file or device ARCHIVE

some-dir

    tar [-] A --catenate --concatenate | c --create | d --diff --compare | --delete | r --append | t --list |
        --test-label | u --update | x --extract --get [options] [pathname ...]

|

    Pipelines
        A  pipeline is a sequence of one or more commands separated by one of the control operators | or |&.  The
        format for a pipeline is:
    
               [time [-p]] [ ! ] command [ [|âŽª|&] command2 ... ]
    
        The standard output of command is connected  via  a  pipe  to  the  standard  input  of  command2.   This
        connection  is performed before any redirections specified by the command (see REDIRECTION below).  If |&
        is used, the standard error of command is connected to command2's standard input through the pipe; it  is
        shorthand  for  2>&1  |.   This  implicit  redirection  of  the  standard  error  is  performed after any
        redirections specified by the command.
    
        The return status of a pipeline is the exit status of the last command, unless  the  pipefail  option  is
        enabled.   If  pipefail  is  enabled,  the  pipeline's return status is the value of the last (rightmost)
        command to exit with a non-zero status, or zero if all commands exit successfully.  If the reserved  word
        !   precedes  a  pipeline, the exit status of that pipeline is the logical negation of the exit status as
        described above.  The shell waits for all commands in the pipeline to terminate before returning a value.
    
        If the time reserved word precedes a pipeline, the elapsed as well as user and system  time  consumed  by
        its execution are reported when the pipeline terminates.  The -p option changes the output format to that
        specified by POSIX.  When the shell is in posix mode, it does not recognize time as a  reserved  word  if
        the  next  token begins with a `-'.  The TIMEFORMAT variable may be set to a format string that specifies
        how the timing information should be displayed; see the description of TIMEFORMAT under  Shell  Variables
        below.
    
        When the shell is in posix mode, time may be followed by a newline.  In this case, the shell displays the
        total user and system time consumed by the shell and its children.  The TIMEFORMAT variable may  be  used
        to specify the format of the time information.
    
        Each command in a pipeline is executed as a separate process (i.e., in a subshell).

ssh

    OpenSSH SSH client (remote login program)

    [ ssh -> https://explainshell.com/explain/1/ssh ]

some-server "cd /; tar xvzf -"

    ssh connects and logs into the specified hostname (with optional user name).  The user must prove his/her
    identity to the remote machine using one of several methods depending on the protocol version used (see
    below).
    
    If command is specified, it is executed on the remote host instead of a login shell. 